org.uqbar.project.wollok/src/org/uqbar/project/wollok/WollokDsl.xtext

```diff
@@ -41,14 +41,17 @@ WPackage:
 // porque a veces xtext no genera bien.
 // eventualmente sería bueno construir estas clases a mano con EMF, y poder
 // agregarles métodos: ej: WNamed.getName() y que las reglas las hereden. Para no necesitar extension methods
-WLibraryElement: (WPackage | WClass | WNamedObject );
-WNamed: WMethodDeclaration | WReferenciable | WPackage | WClass;
-WMethodContainer: WClass | {WObjectLiteral} | WNamedObject;
+WLibraryElement: (WPackage | WClass | WNamedObject | WMixin );
+WNamed: WMethodDeclaration | WReferenciable | WPackage | WClass | WMixin;
+WMethodContainer: WClass | {WObjectLiteral} | WNamedObject | WMixin;
 WReferenciable: WParameter | {WVariable} | WNamedObject ;
 
 WObjectLiteral returns WExpression:
 	{WObjectLiteral}
-	=>'object' '{'
+	=>'object'
+	('inherits' parent=[WClass|QualifiedName] ('(' parentParameters+=WExpression (',' parentParameters+=WExpression)* ')')? )?
+	('mixed with' mixins+=[WMixin|QualifiedName] (('and'|',') mixins+=[WMixin|QualifiedName])*)?
```

Comment:
Well, actually franco didn't like "mixed with". He said that one commonly uses the "mix" verb to refer to the mixing as subject like "Mixin1 mixed in ..." and not the other way.
I'm not sure about that.
He proposed the "includes" which is the same as ruby uses.
I think that both "uses" and "includes" are two ambiguous.
If we follow the same reasoning as we used for changing "extends" to "inherits" we should find a term that matches the concrete case for mixins. That's why I used the "mixed in" phrase, because explicitely tells we talking about mixins.

What if we find another meaning to "uses" or "includes" in the future ?
Also we use to talk about an object "using" another object as collaboration. So it could be confusing.
"Includes" is also used in template engines and other stuff.